flow/activities/maintenance_activity.go

```diff
@@ -0,0 +1,261 @@
+		mirrorStatus, err = a.getMirrorStatus(ctx, mirror)
+		if err != nil {
+			slog.Error("Error querying mirror status for maintenance of previously running mirror",
+				"mirror", mirror.MirrorName, "workflowId", mirror.WorkflowId, "error", err)
+		}
+	}
+	return true, nil
+}
+
+func (a *MaintenanceActivity) CleanBackedUpFlows(ctx context.Context) error {
+	_, err := a.CatalogPool.Exec(ctx, `
+		update maintenance.maintenance_flows
+			set state = $1,
+			restored_at = now(),
+			to_version = $2
+		where state = $3
+	`, mirrorStateRestored, peerdbenv.PeerDBVersionShaShort(), mirrorStateBackup)
+	return err
+}
+
+func (a *MaintenanceActivity) GetBackedUpFlows(ctx context.Context) (*protos.MaintenanceMirrors, error) {
+	rows, err := a.CatalogPool.Query(ctx, `
+		select flow_id, flow_name, workflow_id, flow_created_at, is_cdc
+		from maintenance.maintenance_flows
+		where state = $1
+	`, mirrorStateBackup)
+	if err != nil {
+		return nil, err
+	}
+
+	maintenanceMirrorItems, err := pgx.CollectRows(rows, func(row pgx.CollectableRow) (*protos.MaintenanceMirror, error) {
+		var info protos.MaintenanceMirror
+		err := row.Scan(&info.MirrorId, &info.MirrorName, &info.WorkflowId, &info.MirrorCreatedAt, &info.IsCdc)
+		return &info, err
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &protos.MaintenanceMirrors{
+		Mirrors: maintenanceMirrorItems,
+	}, nil
+}
+
+func (a *MaintenanceActivity) ResumeMirror(ctx context.Context, mirror *protos.MaintenanceMirror) error {
+	mirrorStatus, err := a.getMirrorStatus(ctx, mirror)
+	if err != nil {
+		return err
+	}
+
+	if mirrorStatus != protos.FlowStatus_STATUS_PAUSED {
+		slog.Error("Cannot resume mirror that is not paused",
+			"mirror", mirror.MirrorName, "workflowId", mirror.WorkflowId, "status", mirrorStatus.String())
+		return nil
+	}
+
+	// There can also be "workflow already completed" errors, what should we do in that case?
+	err = model.FlowSignal.SignalClientWorkflow(ctx, a.TemporalClient, mirror.WorkflowId, "", model.NoopSignal)
+	if err != nil {
+		slog.Error("Error signaling mirror to resume for maintenance",
+			"mirror", mirror.MirrorName, "workflowId", mirror.WorkflowId, "error", err)
+		return err
+	}
+	return nil
+}
+
+func (a *MaintenanceActivity) DisableMaintenanceMode(ctx context.Context) error {
+	slog.Info("Disabling maintenance mode")
+	return peerdbenv.UpdatePeerDBMaintenanceModeEnabled(ctx, a.CatalogPool, false)
+}
+
+func (a *MaintenanceActivity) BackgroundAlerter(ctx context.Context) error {
+	defer shared.Interval(ctx, 30*time.Second, func() {
+		activity.RecordHeartbeat(ctx, "Maintenance Workflow is still running")
+	})()
+	defer shared.Interval(ctx, time.Duration(peerdbenv.PeerDBMaintenanceModeWaitAlertSeconds())*time.Second, func() {
+		slog.Warn("Maintenance Workflow is still running")
+		a.Alerter.LogNonFlowWarning(ctx, telemetry.MaintenanceWait, "", "Maintenance mode is still running")
+	})()
+	<-ctx.Done()
```

Comment:
these two can be combined into a single ticker & select in loop. See end of MaintainPull